src/libtrx/game/game_buf.c

```diff
@@ -1,52 +1,28 @@
 #include "game/game_buf.h"
 
 #include "enum_map.h"
-#include "game/shell.h"
-#include "log.h"
 #include "memory.h"
 
-static size_t m_Cap = 0;
-static size_t m_MemUsed = 0;
-static size_t m_MemFree = 0;
-static char *m_MemBase = nullptr;
-static char *m_MemPtr = nullptr;
+static MEMORY_ARENA_ALLOCATOR m_Allocator = {
+    .default_chunk_size = 1024 * 1024 * 5,
+};
 
-void GameBuf_Init(const size_t cap)
+void GameBuf_Init(void)
 {
-    m_Cap = cap;
-    m_MemBase = Memory_Alloc(cap);
-    GameBuf_Reset();
 }
 
 void GameBuf_Reset(void)
 {
-    m_MemPtr = m_MemBase;
-    m_MemFree = m_Cap;
-    m_MemUsed = 0;
+    Memory_ArenaReset(&m_Allocator);
 }
 
 void GameBuf_Shutdown(void)
 {
-    Memory_FreePointer(&m_MemBase);
-    m_Cap = 0;
-    m_MemFree = 0;
-    m_MemUsed = 0;
+    Memory_ArenaFree(&m_Allocator);
 }
 
 void *GameBuf_Alloc(const size_t alloc_size, const GAME_BUFFER buffer)
 {
     const size_t aligned_size = (alloc_size + 3) & ~3;
-
-    const char *const buffer_name = ENUM_MAP_TO_STRING(GAME_BUFFER, buffer);
```

Comment:
I suppose we could get rid of `GAME_BUFFER` altogether now as its only purpose was to log on failure. Or would you prefer to keep it for potential other uses? E.g. maybe logging of memory used with a console command, although that would need it stored somewhere.

Reply:
I was thinking of removing this enum too, but even if we're not using it in the proposed code form, I think keeping it around might prove useful, for instance if we want to check how much memory each buffer occupies.